cmd/mesh/profile-dump.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mesh
+
+import (
+	"io/ioutil"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+
+	"istio.io/operator/pkg/apis/istio/v1alpha2"
+	"istio.io/operator/pkg/component/component"
+	"istio.io/operator/pkg/helm"
+	"istio.io/operator/pkg/tpath"
+	"istio.io/operator/pkg/translate"
+	"istio.io/operator/pkg/util"
+	"istio.io/operator/pkg/validate"
+	"istio.io/operator/pkg/version"
+)
+
+type profileDumpArgs struct {
+	// inFilename is the path to the input IstioControlPlane CR.
+	inFilename string
+	// If set, display the translated Helm values rather than IstioControlPlaneSpec.
+	helmValues bool
+	// rootConfigNode sets the root node for the subtree to display the config for.
+	rootConfigNode string
+	// set is a string with element format "path=value" where path is an IstioControlPlane path and the value is a
+	// value to set the node at that path to.
+	set string
+}
+
+func addProfileDumpFlags(cmd *cobra.Command, args *profileDumpArgs) {
+	cmd.PersistentFlags().StringVarP(&args.inFilename, "filename", "f", "", filenameFlagHelpStr)
+	cmd.PersistentFlags().StringVarP(&args.rootConfigNode, "root-config-node", "r", "",
```

Comment:
this naming is more from internal structure perspective, can we give it a more understandable name to user?

Reply:
yeah, it's not the greatest but i had a hard time thinking of something better. any ideas?

Reply:
The analogous kubectl option is `jsonpath`. In lieu of adopting jsonpath syntax (has this been considered?) we could copy the name, e.g. `path`?

Reply:
I did consider path as a name but it seems too general. How about config-path?
The syntax here follows jsonpath to the extent that it makes sense - there are things that jsonpath allows that wouldn't be supported here so it would be misleading for users for us to say "use jsonpath" and not fully support that syntax. 
I think this is a reasonable compromise - close enough to jsonpath that it will be easy to understand but restrictive enough that all possible expressions are supported. 
One difference where we could be more jsonpath-like is list member selection (a.b.[key:value].c in this path syntax is a.b.[?(@.key==value)].c) but IMO the jsonpath syntax here is just horrible - I mean maybe it makes sense once you use jsonpath as a whole but not for this one case we have.

Reply:
`config-path` is better. 

How often would users need to extra list members with `profile-dump` vs. a named subset (e.g. pilot config only). I suspect the latter would be the primary use case. Aligning with an existing scheme, even if its imperfect, does have advantages in that the user only has to learn one syntax.

Reply:
Probably never - in which case the syntax is the same as jsonpath, so there's nothing new for the user to learn. But the moment we actually call it jsonpath it creates an expectation from the user that they can do jsonpath type of things which we don't support. 
Anyway - I've changed to config-path. I'm open to revisiting the jsonpath question in the future since I do think there's a place for it in operator. Probably more work needed to figure out exactly how and where we could support it. Maybe that will be one of the feedbacks we'll get from users during the alpha.

Reply:
sgtm